src/buildExec.ts

```diff
@@ -46,9 +64,20 @@ const buildExec = () => {
   const slug = core.getInput('slug');
   const swift = core.getInput('swift');
   const swiftProject = core.getInput('swift_project');
-  const token = core.getInput('token');
-  const upstream = core.getInput('upstream_proxy');
   const url = core.getInput('url');
+  let token = core.getInput('token');
+  if (token === '') {
+    let codecovURL = url;
+    if (codecovURL === '') {
+      codecovURL = 'https://codecov.io';
```

Comment:
this should match to `get_config("setup", "codecov_url")` in codecov installation. Looks like correct value for production https://github.com/codecov/codecov-api/blob/main/codecov/settings_prod.py#L33C5-L33C5

Otherwise (enterprise) the URL is defined so we can use that.